csharp/src/Google.Protobuf.Conformance/Program.cs

```diff
@@ -45,6 +47,8 @@ class Program
     {
         private static void Main(string[] args)
         {
+            var mode = (args[0] == "--buffer") ? SerializationMode.Buffer : SerializationMode.Stream;
+
```

Comment:
Are the conformance tests with `--buffer` (or whatever the cmdline arg ends up being) actually being run by the CI?
If not, we should make sure they are being run.

https://github.com/protocolbuffers/protobuf/blob/6263268b8c1b78a8a9b65acd6f5dd5c04dd9b0e1/tests.sh#L148

Reply:
I haven't done anything to set that up. I'm not familiar with the build system. Can you handle this part?

Reply:
I will see what I can do.

Reply:
I added a patch that makes cmdline parsing more robust and enables the --use_buffer_serialization conformance tests (they seems to be failing).